BaikalDataGenerator/Source/render.cpp

```diff
@@ -0,0 +1,437 @@
+#include "XML/tinyxml2.h"
+
+using namespace Baikal;
+
+static std::uint32_t constexpr kNumIterations = 4096;
+
+struct OutputDesc
+{
+    Renderer::OutputType type;
+    std::string name;
+    // extansion of the file to save
+    std::string file_ext;
+    int bpp;
+};
+
+// if you need to add new output for saving to disk
+// just put its description in thic collection
+std::vector<OutputDesc> outputs_collection = { { Renderer::OutputType::kColor, "color", "png", 16 },
+                                               { Renderer::OutputType::kViewShadingNormal, "view_shading_normal", "png", 8 },
+                                               { Renderer::OutputType::kDepth, "depth", "png", 16 },
+                                               { Renderer::OutputType::kAlbedo, "albedo", "png", 8 },
+                                               { Renderer::OutputType::kGloss, "gloss", "png", 8 } };
+
+static bool operator != (RadeonRays::float3 left, RadeonRays::float3 right)
+{
+    return (left.x != right.x) ||
+        (left.y != right.y) ||
+        (left.z != right.z);
+}
+
+Render::Render(const std::string &file_name,
+               const std::string &path,
+               std::uint32_t output_width,
+               std::uint32_t output_height)
+{
+    std::vector<CLWPlatform> platforms;
+    CLWPlatform::CreateAllPlatforms(platforms);
+
+    int platform_index = 0;
+
+    for (auto j = 0u; j < platforms.size(); ++j)
+    {
+        for (auto i = 0u; i < platforms[j].GetDeviceCount(); ++i)
+        {
+            if (platforms[j].GetDevice(i).GetType() == CL_DEVICE_TYPE_GPU)
+            {
+                platform_index = j;
+                break;
+            }
+        }
+    }
+
+    int device_index = 0;
+
+    for (auto i = 0u; i < platforms[platform_index].GetDeviceCount(); ++i)
+    {
+        if (platforms[platform_index].GetDevice(i).GetType() == CL_DEVICE_TYPE_GPU)
+        {
+            device_index = i;
+            break;
+        }
+    }
+
+    auto platform = platforms[platform_index];
+    auto device = platform.GetDevice(device_index);
+    m_context = CLWContext::Create(device);
+
+    m_factory = std::make_unique<Baikal::ClwRenderFactory>(m_context, "cache");
+    m_renderer = m_factory->CreateRenderer(Baikal::ClwRenderFactory::RendererType::kUnidirectionalPathTracer);
+    m_controller = m_factory->CreateSceneController();
+
+    for (const auto output_info: outputs_collection)
+    {
+        m_outputs.push_back(m_factory->CreateOutput(output_width, output_height));
+        m_renderer->SetOutput(output_info.type, m_outputs.back().get());
+    }
+
+    m_scene = (file_name.empty() || path.empty()) ? Baikal::SceneIo::LoadScene("sphere+plane.test", "") :
+                                                    Baikal::SceneIo::LoadScene(file_name, path);
+}
+
+void Render::LoadMaterialXml(const std::string &path, const std::string &file_name)
+{
+    std::stringstream ss;
+    ss << path << "/" << file_name;
```

Comment:
only unix slashes ?

Reply:
I suggest to extract constexpr PATH_DELMITER (delimiter is also used in `tiny_obj_loader.h`).

Reply:
Done